archunit/src/main/java/com/tngtech/archunit/core/importer/AccessRecord.java

```diff
@@ -63,16 +55,31 @@
         AccessType getAccessType();
     }
 
+    @Internal
+    interface CallRecord<T extends CodeUnitCallTarget> extends AccessRecord<T> {
+        Set<JavaClassDescriptor> getCaughtThrowables();
+    }
+
+    @Internal
+    interface MethodCallRecord extends CallRecord<MethodCallTarget> {
+
+    }
+
+    @Internal
+    interface ConstructorCallRecord extends CallRecord<ConstructorCallTarget> {
+
+    }
+
     @Internal
     abstract class Factory<RAW_RECORD, PROCESSED_RECORD> {
 
         abstract PROCESSED_RECORD create(RAW_RECORD record, ImportedClasses classes);
 
-        static Factory<RawAccessRecord, AccessRecord<ConstructorCallTarget>> forConstructorCallRecord() {
-            return new Factory<RawAccessRecord, AccessRecord<ConstructorCallTarget>>() {
+        static Factory<RawAccessRecord.ForMethodCall, ConstructorCallRecord> forConstructorCallRecord() {
+            return new Factory<RawAccessRecord.ForMethodCall, ConstructorCallRecord>() {
                 @Override
-                AccessRecord<ConstructorCallTarget> create(RawAccessRecord record, ImportedClasses classes) {
-                    return new RawAccessRecordProcessed<>(record, classes, CONSTRUCTOR_CALL_TARGET_FACTORY);
+                ConstructorCallRecord create(RawAccessRecord.ForMethodCall record, ImportedClasses classes) {
+                    return new RawConstructorCallRecordProcessed(record, classes);
```

Comment:
Quite a lot of layers to punch through :) I didn't quite get the idea behind `RawAccessRecord`/`RawXXXRecordProcessed`

Reply:
I can explain the idea quickly: `RawAccessRecord` represents the raw data as collected during bytecode analysis (i.e. origin and target are just strings or type descriptors, etc.). `RawAccessRecordProcessed` represents the next step, when this raw data has been resolved to domain objects (e.g. the target is now not a string anymore, but a `JavaMethodCallTarget` which has a `JavaClass` as its owner, the origin is a `JavaCodeUnit`, etc.).